pilot/pkg/proxy/envoy/v2/eds.go

```diff
@@ -72,7 +72,7 @@ var (
 // all clusters until we support on-demand cluster loading.
 type EdsCluster struct {
 	// mutex protects changes to this cluster
-	mutex sync.Mutex
+	mutex sync.RWMutex
```

Comment:
you do realize that we literally made the same comment (use rw mutex instead of mutex) when you first wrote the incremental EDS, right?
:). Better late than never

Reply:
Yes, I remembered the comment and updated it - won't make a huge difference.

The problem is 'defer mutex.Unlock' and having method calls with a mutex held.